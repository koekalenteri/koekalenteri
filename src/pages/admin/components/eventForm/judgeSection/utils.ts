import type { DeepPartial, EventClass, EventType, Judge, PublicJudge } from '../../../../../types'
import type { PartialEvent } from '../../EventForm'

import { isSameDay } from 'date-fns'

export type PartialPublicJudge = Partial<PublicJudge>
export type PartialPublicJudgeValue = PartialPublicJudge | PartialPublicJudge[]

export const filterJudges = (
  judges: Judge[],
  eventJudges: PublicJudge[],
  id: number | undefined,
  eventType?: EventType
) =>
  judges
    .filter((j) => !eventType?.official || j.eventTypes.includes(eventType.eventType))
    .filter((j) => j.id === id || !eventJudges.find((ej) => ej.id === j.id))

export const hasJudge = (c: DeepPartial<EventClass>, id?: number) =>
  Array.isArray(c.judge) ? c.judge.find((j) => j.id === id) : c.judge?.id === id

export const filterClassesByJudgeId = (classes?: EventClass[], id?: number) => classes?.filter((c) => hasJudge(c, id))

export const toArray = (j?: PartialPublicJudge): PartialPublicJudge[] => (j ? [j] : [])
export const makeArray = (j?: PartialPublicJudgeValue) => (Array.isArray(j) ? [...j] : toArray(j))
export const selectJudge = (j?: PartialPublicJudgeValue, judge?: PublicJudge): PartialPublicJudge[] => {
  const a = makeArray(j)
  if (judge) {
    const index = a.findIndex((cj) => cj.id === judge.id)
    if (index === -1) {
      a.push(judge)
    } else {
      a[index] = judge
    }
  }
  return a
}

export const removeJudge = (j?: PartialPublicJudgeValue, id?: number): PartialPublicJudge[] => {
  const a = makeArray(j)
  return a.filter((cj) => cj.id !== id)
}

export const updateJudge = (
  event: PartialEvent,
  id: number | undefined,
  judge: PublicJudge | undefined,
  values?: DeepPartial<EventClass>[]
) => {
  const isSelected = (c: DeepPartial<EventClass>) =>
    values?.find((v) => event && isSameDay(v.date ?? event.startDate, c.date ?? event.startDate) && v.class === c.class)
  return event.classes?.map((c) => ({
    ...c,
    judge: isSelected(c) ? selectJudge(c.judge, judge) : removeJudge(c.judge, id),
  }))
}
