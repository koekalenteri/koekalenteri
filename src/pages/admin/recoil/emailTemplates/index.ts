export * from './actions'
export * from './atoms'
export * from './atoms2'
export * from './effects'
export * from './selectors'
