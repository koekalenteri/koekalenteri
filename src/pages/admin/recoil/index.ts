export * from './emailTemplates'
export * from './events'
export * from './eventTypes'
export * from './officials'
export * from './organizers'
export * from './registrations'
export * from './user'
