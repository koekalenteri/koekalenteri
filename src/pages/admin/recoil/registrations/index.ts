export * from './atoms'
// 2nd file to avoid circular deps
export * from './atoms2'
export * from './selectors'
