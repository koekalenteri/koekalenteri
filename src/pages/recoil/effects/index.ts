export * from './log'
export * from './storage'

export const getParamFromFamilyKey = (key: string) => key.split('__')[1]?.slice(1, -1)
