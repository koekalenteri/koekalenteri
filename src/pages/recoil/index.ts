export * from './effects'
export * from './events'
export * from './judges'
export * from './registration'
export * from './user'
