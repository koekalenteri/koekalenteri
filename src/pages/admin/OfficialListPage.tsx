import type { Theme } from '@mui/material'
import type { GridColDef } from '@mui/x-data-grid'
import type { Official } from '../../types'

import { useTranslation } from 'react-i18next'
import CloudSync from '@mui/icons-material/CloudSync'
import { useMediaQuery } from '@mui/material'
import Button from '@mui/material/Button'
import Stack from '@mui/material/Stack'
import { useRecoilState, useRecoilValue } from 'recoil'

import StyledDataGrid from '../components/StyledDataGrid'
import { isAdminSelector } from '../recoil'

import FullPageFlex from './components/FullPageFlex'
import { QuickSearchToolbar } from './components/QuickSearchToolbar'
import { filteredOfficialsSelector, officialFilterAtom, useOfficialsActions } from './recoil'

declare module '@mui/x-data-grid' {
  interface ToolbarPropsOverrides {
    value: string
    onChange: (event: React.ChangeEvent<HTMLInputElement>) => void
    clearSearch: () => void
  }
}

export default function OfficialListPage() {
  const large = useMediaQuery((theme: Theme) => theme.breakpoints.up('md'))
  const [searchText, setSearchText] = useRecoilState(officialFilterAtom)
  const { t } = useTranslation()
  const officials = useRecoilValue(filteredOfficialsSelector)
  const isAdmin = useRecoilValue(isAdminSelector)
  const actions = useOfficialsActions()

  const columns: GridColDef<Official>[] = [
    {
      field: 'name',
      flex: 1,
      headerName: t('name'),
      minWidth: 150,
    },
    {
      field: 'id',
      flex: 0,
      headerName: t('id'),
      width: 80,
    },
    {
      field: 'location',
      flex: 0,
      headerName: t('contact.city'),
      width: 120,
    },
    {
      field: 'phone',
      flex: 0,
      headerName: t('contact.phone'),
      width: 150,
    },
    {
      field: 'email',
      flex: 1,
      headerName: t('contact.email'),
      minWidth: 150,
    },
    {
      field: 'district',
      flex: 1,
      headerName: t('district'),
    },
    {
      field: 'eventTypes',
      flex: 1,
      headerName: t('eventTypes'),
      valueGetter: (value: Official['eventTypes']) => value?.join(', '),
    },
  ]

  return (
    <FullPageFlex>
      <Stack direction="row" spacing={2}>
        <Button startIcon={<CloudSync />} onClick={actions.refresh} sx={{ display: isAdmin ? undefined : 'none' }}>
          {t('updateData', { data: 'officials' })}
        </Button>
      </Stack>

      <StyledDataGrid
        columns={columns}
        columnVisibilityModel={{
          district: large,
          eventTypes: large,
          id: large,
          location: large,
        }}
        slots={{ toolbar: QuickSearchToolbar }}
        slotProps={{
          toolbar: {
            value: searchText,
            onChange: (event: React.ChangeEvent<HTMLInputElement>) => setSearchText(event.target.value),
            clearSearch: () => setSearchText(''),
          },
        }}
        rows={officials}
      />
    </FullPageFlex>
  )
}
