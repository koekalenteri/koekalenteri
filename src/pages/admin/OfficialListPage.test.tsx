import { Suspense } from 'react'
import { MemoryRouter } from 'react-router-dom'
import { ThemeProvider } from '@mui/material'
import { render } from '@testing-library/react'
import { SnackbarProvider } from 'notistack'
import { RecoilRoot } from 'recoil'

import theme from '../../assets/Theme'
import { flushPromises } from '../../test-utils/utils'

import OfficialListPage from './OfficialListPage'

jest.mock('../../api/official')
jest.mock('../../api/user')

describe('OfficialListPage', () => {
  beforeAll(() => jest.useFakeTimers())
  afterEach(() => jest.clearAllTimers())
  afterAll(() => jest.useRealTimers())

  it('renders', async () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <RecoilRoot>
          <MemoryRouter>
            <Suspense fallback={<div>loading...</div>}>
              <SnackbarProvider>
                <OfficialListPage />
              </SnackbarProvider>
            </Suspense>
          </MemoryRouter>
        </RecoilRoot>
      </ThemeProvider>
    )
    await flushPromises()
    expect(container).toMatchSnapshot()
  })
})
