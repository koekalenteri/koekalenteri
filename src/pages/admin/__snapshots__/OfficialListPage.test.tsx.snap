// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OfficialListPage renders 1`] = `
<div>
  <div
    class="MuiBox-root css-1uovp08"
  >
    <div
      class="MuiBox-root css-10784zt"
    >
      <div
        class="MuiStack-root css-1hezniw-MuiStack-root"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-ibqjzf-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-icon MuiButton-startIcon MuiButton-iconSizeMedium css-1sh91j5-MuiButton-startIcon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
              data-testid="CloudSyncIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M21.5 14.98c-.02 0-.03 0-.05.01C21.2 13.3 19.76 12 18 12c-1.4 0-2.6.83-3.16 2.02C13.26 14.1 12 15.4 12 17c0 1.66 1.34 3 3 3l6.5-.02c1.38 0 2.5-1.12 2.5-2.5s-1.12-2.5-2.5-2.5M10 4.26v2.09C7.67 7.18 6 9.39 6 12c0 1.77.78 3.34 2 4.44V14h2v6H4v-2h2.73C5.06 16.54 4 14.4 4 12c0-3.73 2.55-6.85 6-7.74M20 6h-2.73c1.43 1.26 2.41 3.01 2.66 5h-2.02c-.23-1.36-.93-2.55-1.91-3.44V10h-2V4h6z"
              />
            </svg>
          </span>
          updateData data
        </button>
      </div>
      <div
        class="MuiDataGrid-root MuiDataGrid-root--densityCompact MuiDataGrid-withBorderColor css-1lu635c-MuiDataGrid-root"
        style="--DataGrid-hasScrollX: 1; --DataGrid-hasScrollY: 1; --DataGrid-scrollbarSize: 0px; --DataGrid-rowWidth: 150px; --DataGrid-columnsTotalWidth: 150px; --DataGrid-leftPinnedWidth: 0px; --DataGrid-rightPinnedWidth: 0px; --DataGrid-headerHeight: 39px; --DataGrid-headersTotalHeight: 39px; --DataGrid-topContainerHeight: 39px; --DataGrid-bottomContainerHeight: 0px; --height: 28px;"
      >
        <div
          class="MuiStack-root css-k5y4k9-MuiStack-root"
        >
          <div
            class="MuiDataGrid-toolbarContainer css-1wrr6qg-MuiDataGrid-toolbarContainer"
          >
            <div
              class="MuiFormControl-root MuiTextField-root css-zl6cqu-MuiFormControl-root-MuiTextField-root"
            >
              <div
                class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedStart MuiInputBase-adornedEnd css-1624vf1-MuiInputBase-root-MuiInput-root"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-120dh41-MuiSvgIcon-root"
                  data-testid="SearchIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14"
                  />
                </svg>
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedStart MuiInputBase-inputAdornedEnd css-1yrc8ca-MuiInputBase-input-MuiInput-input"
                  id=":r1:"
                  placeholder="search.placeholder"
                  type="text"
                  value=""
                />
                <button
                  aria-label="clear"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-155wbr6-MuiButtonBase-root-MuiIconButton-root"
                  style="visibility: hidden;"
                  tabindex="0"
                  title="clear"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-120dh41-MuiSvgIcon-root"
                    data-testid="ClearIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                    />
                  </svg>
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          aria-colcount="3"
          aria-multiselectable="false"
          aria-rowcount="4"
          class="MuiDataGrid-main css-1bgckiu-MuiDataGrid-main"
          role="grid"
          tabindex="-1"
        >
          <div
            class="css-18l11qw"
            data-id="gridPanelAnchor"
            role="presentation"
          />
          <div
            class="MuiDataGrid-virtualScroller MuiDataGrid-virtualScroller--hasScrollX css-1vigakz-MuiDataGrid-virtualScroller"
            role="presentation"
            style="overflow-x: hidden;"
          >
            <div
              class="MuiDataGrid-topContainer MuiDataGrid-container--top css-b1wygl"
              role="presentation"
            >
              <div
                class="MuiDataGrid-columnHeaders css-1gtv474-MuiDataGrid-columnHeaders"
                role="presentation"
              >
                <div
                  aria-rowindex="1"
                  class="MuiDataGrid-row--borderBottom css-yseucu-MuiDataGrid-columnHeaderRow"
                  role="row"
                >
                  <div
                    role="presentation"
                    style="width: 0px;"
                  />
                  <div
                    aria-colindex="1"
                    aria-sort="none"
                    class="MuiDataGrid-columnHeader MuiDataGrid-columnHeader--sortable MuiDataGrid-withBorderColor"
                    data-field="name"
                    role="columnheader"
                    style="height: 39px; width: 0px;"
                    tabindex="0"
                  >
                    <div
                      class="MuiDataGrid-columnHeaderDraggableContainer"
                      draggable="false"
                      role="presentation"
                    >
                      <div
                        class="MuiDataGrid-columnHeaderTitleContainer"
                        role="presentation"
                      >
                        <div
                          class="MuiDataGrid-columnHeaderTitleContainerContent"
                        >
                          <div
                            class="MuiDataGrid-columnHeaderTitle css-1gqmilo-MuiDataGrid-columnHeaderTitle"
                            data-mui-internal-clone-element="true"
                          >
                            name
                          </div>
                        </div>
                        <div
                          class="MuiDataGrid-iconButtonContainer css-ltf0zy-MuiDataGrid-iconButtonContainer"
                        >
                          <button
                            aria-label="Sort"
                            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall MuiDataGrid-sortButton css-155wbr6-MuiButtonBase-root-MuiIconButton-root"
                            field="name"
                            tabindex="-1"
                            title="Sort"
                            type="button"
                          >
                            <svg
                              aria-hidden="true"
                              class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall MuiDataGrid-sortIcon css-120dh41-MuiSvgIcon-root"
                              data-testid="ArrowUpwardIcon"
                              focusable="false"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
                              />
                            </svg>
                          </button>
                        </div>
                      </div>
                    </div>
                    <div
                      class="MuiDataGrid-columnSeparator MuiDataGrid-columnSeparator--resizable MuiDataGrid-columnSeparator--sideRight"
                      style="min-height: 39px;"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiDataGrid-iconSeparator css-1umw9bq-MuiSvgIcon-root"
                        data-testid="SeparatorIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <rect
                          height="24"
                          rx="0.5"
                          width="1"
                          x="11.5"
                        />
                      </svg>
                    </div>
                  </div>
                  <div
                    aria-colindex="2"
                    aria-sort="none"
                    class="MuiDataGrid-columnHeader MuiDataGrid-columnHeader--sortable MuiDataGrid-withBorderColor"
                    data-field="phone"
                    role="columnheader"
                    style="height: 39px; width: 150px;"
                    tabindex="-1"
                  >
                    <div
                      class="MuiDataGrid-columnHeaderDraggableContainer"
                      draggable="false"
                      role="presentation"
                    >
                      <div
                        class="MuiDataGrid-columnHeaderTitleContainer"
                        role="presentation"
                      >
                        <div
                          class="MuiDataGrid-columnHeaderTitleContainerContent"
                        >
                          <div
                            class="MuiDataGrid-columnHeaderTitle css-1gqmilo-MuiDataGrid-columnHeaderTitle"
                            data-mui-internal-clone-element="true"
                          >
                            contact.phone
                          </div>
                        </div>
                        <div
                          class="MuiDataGrid-iconButtonContainer css-ltf0zy-MuiDataGrid-iconButtonContainer"
                        >
                          <button
                            aria-label="Sort"
                            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall MuiDataGrid-sortButton css-155wbr6-MuiButtonBase-root-MuiIconButton-root"
                            field="phone"
                            tabindex="-1"
                            title="Sort"
                            type="button"
                          >
                            <svg
                              aria-hidden="true"
                              class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall MuiDataGrid-sortIcon css-120dh41-MuiSvgIcon-root"
                              data-testid="ArrowUpwardIcon"
                              focusable="false"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
                              />
                            </svg>
                          </button>
                        </div>
                      </div>
                    </div>
                    <div
                      class="MuiDataGrid-columnSeparator MuiDataGrid-columnSeparator--resizable MuiDataGrid-columnSeparator--sideRight"
                      style="min-height: 39px;"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiDataGrid-iconSeparator css-1umw9bq-MuiSvgIcon-root"
                        data-testid="SeparatorIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <rect
                          height="24"
                          rx="0.5"
                          width="1"
                          x="11.5"
                        />
                      </svg>
                    </div>
                  </div>
                  <div
                    aria-colindex="3"
                    aria-sort="none"
                    class="MuiDataGrid-columnHeader MuiDataGrid-columnHeader--sortable MuiDataGrid-withBorderColor MuiDataGrid-columnHeader--lastUnpinned MuiDataGrid-columnHeader--last"
                    data-field="email"
                    role="columnheader"
                    style="height: 39px; width: 0px;"
                    tabindex="-1"
                  >
                    <div
                      class="MuiDataGrid-columnHeaderDraggableContainer"
                      draggable="false"
                      role="presentation"
                    >
                      <div
                        class="MuiDataGrid-columnHeaderTitleContainer"
                        role="presentation"
                      >
                        <div
                          class="MuiDataGrid-columnHeaderTitleContainerContent"
                        >
                          <div
                            class="MuiDataGrid-columnHeaderTitle css-1gqmilo-MuiDataGrid-columnHeaderTitle"
                            data-mui-internal-clone-element="true"
                          >
                            contact.email
                          </div>
                        </div>
                        <div
                          class="MuiDataGrid-iconButtonContainer css-ltf0zy-MuiDataGrid-iconButtonContainer"
                        >
                          <button
                            aria-label="Sort"
                            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall MuiDataGrid-sortButton css-155wbr6-MuiButtonBase-root-MuiIconButton-root"
                            field="email"
                            tabindex="-1"
                            title="Sort"
                            type="button"
                          >
                            <svg
                              aria-hidden="true"
                              class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall MuiDataGrid-sortIcon css-120dh41-MuiSvgIcon-root"
                              data-testid="ArrowUpwardIcon"
                              focusable="false"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
                              />
                            </svg>
                          </button>
                        </div>
                      </div>
                    </div>
                    <div
                      class="MuiDataGrid-columnSeparator MuiDataGrid-columnSeparator--resizable MuiDataGrid-columnSeparator--sideRight"
                      style="min-height: 39px;"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiDataGrid-iconSeparator css-1umw9bq-MuiSvgIcon-root"
                        data-testid="SeparatorIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <rect
                          height="24"
                          rx="0.5"
                          width="1"
                          x="11.5"
                        />
                      </svg>
                    </div>
                  </div>
                  <div
                    class="MuiDataGrid-filler"
                    role="presentation"
                  />
                  <div
                    class="MuiDataGrid-scrollbarFiller MuiDataGrid-scrollbarFiller--header"
                    role="presentation"
                  />
                </div>
              </div>
            </div>
            <div
              class="MuiDataGrid-virtualScrollerContent css-aymtem-MuiDataGrid-virtualScrollerContent"
              role="presentation"
              style="width: auto; flex-shrink: 0; flex-basis: 84px;"
            >
              <div
                class="MuiDataGrid-virtualScrollerRenderZone css-11dqcl8-MuiDataGrid-virtualScrollerRenderZone"
                role="rowgroup"
                style="transform: translate3d(0, 0px, 0);"
              >
                <div
                  aria-rowindex="2"
                  aria-selected="false"
                  class="MuiDataGrid-row MuiDataGrid-row--firstVisible"
                  data-id="333"
                  data-rowindex="0"
                  role="row"
                  style="max-height: 28px; min-height: 28px; --height: 28px;"
                >
                  <div
                    class="MuiDataGrid-cellOffsetLeft"
                    role="presentation"
                    style="width: 0px;"
                  />
                  <div
                    aria-colindex="1"
                    aria-colspan="1"
                    aria-rowspan="1"
                    class="MuiDataGrid-cell MuiDataGrid-cell--textLeft"
                    data-colindex="0"
                    data-field="name"
                    role="gridcell"
                    style="--width: 0px;"
                    tabindex="-1"
                    title="Toimitsija 1"
                  >
                    Toimitsija 1
                  </div>
                  <div
                    aria-colindex="2"
                    aria-colspan="1"
                    aria-rowspan="1"
                    class="MuiDataGrid-cell MuiDataGrid-cell--textLeft"
                    data-colindex="1"
                    data-field="phone"
                    role="gridcell"
                    style="--width: 150px;"
                    tabindex="-1"
                    title="123456781"
                  >
                    123456781
                  </div>
                  <div
                    aria-colindex="3"
                    aria-colspan="1"
                    aria-rowspan="1"
                    class="MuiDataGrid-cell MuiDataGrid-cell--textLeft"
                    data-colindex="2"
                    data-field="email"
                    role="gridcell"
                    style="--width: 0px;"
                    tabindex="-1"
                    title="toimari1@sposti.not"
                  >
                    toimari1@sposti.not
                  </div>
                  <div
                    class="MuiDataGrid-cell MuiDataGrid-cellEmpty"
                    role="presentation"
                  />
                </div>
                <div
                  aria-rowindex="3"
                  aria-selected="false"
                  class="MuiDataGrid-row"
                  data-id="444"
                  data-rowindex="1"
                  role="row"
                  style="max-height: 28px; min-height: 28px; --height: 28px;"
                >
                  <div
                    class="MuiDataGrid-cellOffsetLeft"
                    role="presentation"
                    style="width: 0px;"
                  />
                  <div
                    aria-colindex="1"
                    aria-colspan="1"
                    aria-rowspan="1"
                    class="MuiDataGrid-cell MuiDataGrid-cell--textLeft"
                    data-colindex="0"
                    data-field="name"
                    role="gridcell"
                    style="--width: 0px;"
                    tabindex="-1"
                    title="Toimitsija 2"
                  >
                    Toimitsija 2
                  </div>
                  <div
                    aria-colindex="2"
                    aria-colspan="1"
                    aria-rowspan="1"
                    class="MuiDataGrid-cell MuiDataGrid-cell--textLeft"
                    data-colindex="1"
                    data-field="phone"
                    role="gridcell"
                    style="--width: 150px;"
                    tabindex="-1"
                    title="123456782"
                  >
                    123456782
                  </div>
                  <div
                    aria-colindex="3"
                    aria-colspan="1"
                    aria-rowspan="1"
                    class="MuiDataGrid-cell MuiDataGrid-cell--textLeft"
                    data-colindex="2"
                    data-field="email"
                    role="gridcell"
                    style="--width: 0px;"
                    tabindex="-1"
                    title="toimari2@sposti.not"
                  >
                    toimari2@sposti.not
                  </div>
                  <div
                    class="MuiDataGrid-cell MuiDataGrid-cellEmpty"
                    role="presentation"
                  />
                </div>
                <div
                  aria-rowindex="4"
                  aria-selected="false"
                  class="MuiDataGrid-row MuiDataGrid-row--lastVisible"
                  data-id="555"
                  data-rowindex="2"
                  role="row"
                  style="max-height: 28px; min-height: 28px; --height: 28px;"
                >
                  <div
                    class="MuiDataGrid-cellOffsetLeft"
                    role="presentation"
                    style="width: 0px;"
                  />
                  <div
                    aria-colindex="1"
                    aria-colspan="1"
                    aria-rowspan="1"
                    class="MuiDataGrid-cell MuiDataGrid-cell--textLeft"
                    data-colindex="0"
                    data-field="name"
                    role="gridcell"
                    style="--width: 0px;"
                    tabindex="-1"
                    title="Toimitsija 3"
                  >
                    Toimitsija 3
                  </div>
                  <div
                    aria-colindex="2"
                    aria-colspan="1"
                    aria-rowspan="1"
                    class="MuiDataGrid-cell MuiDataGrid-cell--textLeft"
                    data-colindex="1"
                    data-field="phone"
                    role="gridcell"
                    style="--width: 150px;"
                    tabindex="-1"
                    title="123456783"
                  >
                    123456783
                  </div>
                  <div
                    aria-colindex="3"
                    aria-colspan="1"
                    aria-rowspan="1"
                    class="MuiDataGrid-cell MuiDataGrid-cell--textLeft"
                    data-colindex="2"
                    data-field="email"
                    role="gridcell"
                    style="--width: 0px;"
                    tabindex="-1"
                    title="toimari3@sposti.not"
                  >
                    toimari3@sposti.not
                  </div>
                  <div
                    class="MuiDataGrid-cell MuiDataGrid-cellEmpty"
                    role="presentation"
                  />
                </div>
              </div>
            </div>
            <div
              class="MuiDataGrid-bottomContainer MuiDataGrid-container--bottom css-gzxaox"
              role="presentation"
            />
          </div>
          <div
            aria-hidden="true"
            class="MuiDataGrid-scrollbar MuiDataGrid-scrollbar--horizontal css-2n9dey"
            tabindex="-1"
          >
            <div
              class="MuiDataGrid-scrollbarContent"
            />
          </div>
          <div
            aria-hidden="true"
            class="MuiDataGrid-scrollbar MuiDataGrid-scrollbar--vertical css-1ir2opr"
            tabindex="-1"
          >
            <div
              class="MuiDataGrid-scrollbarContent"
            />
          </div>
        </div>
        <div
          class="MuiDataGrid-footerContainer MuiDataGrid-withBorderColor css-15lx25q-MuiDataGrid-footerContainer"
        >
          <div />
          <div
            class="MuiTablePagination-root css-1hr2sou-MuiTablePagination-root"
          >
            <div
              class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular MuiTablePagination-toolbar css-1gak8h1-MuiToolbar-root-MuiTablePagination-toolbar"
            >
              <div
                class="MuiTablePagination-spacer css-1wtxofq-MuiTablePagination-spacer"
              />
              <p
                class="MuiTablePagination-selectLabel css-s09cke-MuiTablePagination-selectLabel"
                id=":r4:"
              >
                Rows per page:
              </p>
              <div
                class="MuiInputBase-root MuiInputBase-colorPrimary MuiTablePagination-select MuiSelect-root MuiTablePagination-input css-oegngy-MuiInputBase-root-MuiTablePagination-select"
              >
                <div
                  aria-expanded="false"
                  aria-haspopup="listbox"
                  aria-labelledby=":r4: :r3:"
                  class="MuiSelect-select MuiTablePagination-select MuiSelect-standard MuiInputBase-input css-mmygx2-MuiSelect-select-MuiInputBase-input"
                  id=":r3:"
                  role="combobox"
                  tabindex="0"
                >
                  100
                </div>
                <input
                  aria-hidden="true"
                  aria-invalid="false"
                  class="MuiSelect-nativeInput css-j0riat-MuiSelect-nativeInput"
                  tabindex="-1"
                  value="100"
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiTablePagination-selectIcon MuiSelect-iconStandard css-1gpuzre-MuiSvgIcon-root-MuiSelect-icon"
                  data-testid="ArrowDropDownIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
              </div>
              <p
                class="MuiTablePagination-displayedRows css-11cfq65-MuiTablePagination-displayedRows"
              >
                1–3 of 3
              </p>
              <div
                class="MuiTablePagination-actions"
              >
                <button
                  aria-label="Go to previous page"
                  class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-h7cjts-MuiButtonBase-root-MuiIconButton-root"
                  disabled=""
                  tabindex="-1"
                  title="Go to previous page"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                    data-testid="KeyboardArrowLeftIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                    />
                  </svg>
                </button>
                <button
                  aria-label="Go to next page"
                  class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-h7cjts-MuiButtonBase-root-MuiIconButton-root"
                  disabled=""
                  tabindex="-1"
                  title="Go to next page"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                    data-testid="KeyboardArrowRightIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                    />
                  </svg>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
